relay-quotas/src/rate_limit.rs

```diff
@@ -156,8 +157,11 @@ pub struct RateLimit {
     /// A marker when this rate limit expires.
     pub retry_after: RetryAfter,
 
-    /// The namespace of this rate limit.
-    pub namespace: Option<MetricNamespace>,
+    /// The metric namespace of this rate limit.
+    ///
+    /// Ignored on all data categories except for `MetricBucket`. If empty, this rate limit applies
+    /// to metrics of all namespaces.
+    pub namespace: SmallVec<[MetricNamespace; 1]>,
```

Comment:
This field is not part of any protocol, right? So we could rename it.
```suggestion
    pub namespaces: SmallVec<[MetricNamespace; 1]>,
```